crates/tendermint/src/chain/id.rs

```diff
@@ -0,0 +1,147 @@
+//! Tendermint blockchain identifiers
+
+#[cfg(feature = "serializers")]
+use serde::{de::Error as DeError, Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    cmp::Ordering,
+    fmt::{self, Debug, Display},
+    hash::{Hash, Hasher},
+    str::{self, FromStr},
+};
+
+use error::Error;
+
+/// Maximum length of a `chain::Id` name
+pub const MAX_LENGTH: usize = 64;
+
+/// Chain identifier (e.g. 'gaia-9000')
+#[derive(Copy, Clone)]
+pub struct Id([u8; MAX_LENGTH]);
```

Comment:
Uses a `\0`-terminated byte array as the backing storage for the `Id`, which lets this type derive `Copy`, which I think is pretty nice for something like this (as opposed to using heap-backed `String`s and `clone()`)